apiserver/facades/controller/charmrevisionupdater/charmhub.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// TODO(benhoyt) - also add caching and retries, like we do with charmstore
+
+package charmrevisionupdater
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/juju/charm/v8/resource"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/charmhub"
+	"github.com/juju/juju/charmhub/transport"
+)
+
+// charmhubID holds identifying information for several charms for a
+// charmhubLatestCharmInfo call.
+type charmhubID struct {
+	id       string
+	revision int
+	channel  string
+	os       string
+	series   string
+	arch     string
+}
+
+// charmhubResult is the type charmhubLatestCharmInfo returns: information
+// about a charm revision and its resources.
+type charmhubResult struct {
+	name      string
+	timestamp time.Time
+	revision  int
+	resources []resource.Resource
+	error     error
+}
+
+// CharmhubRefreshClient is an interface for the methods of the charmhub
+// client that we need.
+type CharmhubRefreshClient interface {
+	Refresh(ctx context.Context, config charmhub.RefreshConfig) ([]transport.RefreshResponse, error)
+}
+
+// charmhubLatestCharmInfo fetches the latest information about the given
+// charms from charmhub's "charm_refresh" API.
+func charmhubLatestCharmInfo(client CharmhubRefreshClient, ids []charmhubID) ([]charmhubResult, error) {
+	cfgs := make([]charmhub.RefreshConfig, len(ids))
+	for i, id := range ids {
+		platform := charmhub.RefreshPlatform{
+			Architecture: id.arch,
+			OS:           id.os,
+			Series:       id.series,
+		}
+		cfg, err := charmhub.RefreshOne(id.id, id.revision, id.channel, platform)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		logger.Infof("TODO charmhubLatestCharmInfo cfg %d = %#v", i, cfg)
```

Comment:
Usually we do something like
// TODO <username> <date>
// charmhubLatestCharmInfo cfg %d = %#v", i, cfg

If it gets left behind for any reason, it doesn't bleed out to the users.